### ISSUE TITLE
"Server identity cannot be verified" after update

### ISSUE THREAD

**Issue by sokratis1988** [mailcow/mailcow-dockerized#3903]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [ X] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [X ] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X ] I have understood that answers are voluntary and community-driven, and not commercial support.
- [X ] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary

Yesterday I updated my Mailcow-Server and after that I was not able to check my mails anymore. I get the error "Server identity cannot be verified". As certificate/domain name the message contains mail.my-domain.com.
When I click on details I see the certificate name "autoconfig.my-other-domain.com". Not trusted Valid until 18.03.2021.

I have more domains and all are listed on this certificate under alternative domains of issuer.
All of them are for autoconfig.x.com

I also tried to force renew certificate but this is not working - I dont know why. The logs are empty - no try to renew the certificates.

How I "fix" the issue temporarily:
I copied from an old backup "mail.my-domain.com" the cert and key in data/assets/ssl and restart dovecote, nginx, postfix.
Now I can check my mails!

This should be fixed. I don't know what the real problem is - but there is something...

## System information

| Question | Answer |
| --- | --- |
| My operating system | Ubuntu |
| Is Apparmor, SELinux or similar active? | No |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | No information |
| Server/VM specifications (Memory, CPU Cores) |2vCPU |
| Docker Version (`docker version`) | 20.10.1 |
| Docker-Compose Version (`docker-compose version`) | 1.27.4, build 40524192 |
| Reverse proxy (custom solution) | No |

- Output of `git diff origin/master`, any other changes to the code? If so, **please post them**.
Just hostname changes
- All third-party firewalls and custom iptables rules are unsupported. *Please check the Docker docs about how to use Docker with your own ruleset*. Nevertheless, iptabels output can help us to help you: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat`.
- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output.

**Comment by andryyy** [comment]
Hi,

How is this a bug without any logs? Sounds like a DNS issue or something else. "This should be fixed" => What should be fixed?

**Comment by sokratis1988** [comment]
There was no change on DNS-site. As I mentioned, it was working fine before the update and also working now with the old certificates from the backup.

> cd /opt/mailcow-dockerized
>touch data/assets/ssl/force_renew
>docker-compose restart acme-mailcow
>docker-compose logs --tail=200 -f acme-mailcow

The renew is not working. No logs for renewal, no deleting of the file

**Comment by andryyy** [comment]
I don't see a bug. Please add logs and a way to reproduce said bug. acme-mailcow works fine on every installation we updated in our farm. :) 

You can find us on Telegram and community.mailcow.email if you need support.

**Comment by billsuxx** [comment]
I have the same issue. Updated  the server, restarted all docker containers, and now the iOS devices shows the "Cannot Verify Server Identity" popup all the time. No DNS or configuration changes.

**Comment by andryyy** [comment]
Update mailcow. That can happen when you update to Docker 20.x and forget to update mailcow.

And again: **we cannot help without logs!!!**

**Comment by sokratis1988** [comment]
@andryyy 
When I talk about "update" I mean the update of mailcow!
What else can I do?

What logs do you need? from which container?

```
# docker-compose logs --tail=200 -f acme-mailcow
acme-mailcow_1       | Fri Dec 18 10:32:50 CET 2020 - Waiting for Docker API...
acme-mailcow_1       | Fri Dec 18 10:32:50 CET 2020 - Docker API OK
acme-mailcow_1       | Fri Dec 18 10:32:50 CET 2020 - Waiting for Postfix...
acme-mailcow_1       | Fri Dec 18 10:32:50 CET 2020 - Postfix OK
acme-mailcow_1       | Fri Dec 18 10:32:50 CET 2020 - Waiting for Dovecot...
acme-mailcow_1       | Fri Dec 18 10:32:51 CET 2020 - Dovecot OK
acme-mailcow_1       | Fri Dec 18 10:32:51 CET 2020 - Waiting for database...
acme-mailcow_1       | Fri Dec 18 10:32:51 CET 2020 - Database OK
acme-mailcow_1       | Fri Dec 18 10:32:51 CET 2020 - Waiting for Nginx...
acme-mailcow_1       | Fri Dec 18 10:37:19 CET 2020 - Waiting for Docker API...
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Docker API OK
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Waiting for Postfix...
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Postfix OK
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Waiting for Dovecot...
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Dovecot OK
acme-mailcow_1       | Fri Dec 18 10:37:20 CET 2020 - Waiting for database...
acme-mailcow_1       | Fri Dec 18 10:37:24 CET 2020 - Database OK
acme-mailcow_1       | Fri Dec 18 10:37:24 CET 2020 - Waiting for Nginx...
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Waiting for Docker API...
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Docker API OK
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Waiting for Postfix...
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Postfix OK
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Waiting for Dovecot...
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Dovecot OK
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Waiting for database...
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Database OK
acme-mailcow_1       | Fri Dec 18 10:41:00 CET 2020 - Waiting for Nginx...
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Waiting for Docker API...
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Docker API OK
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Waiting for Postfix...
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Postfix OK
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Waiting for Dovecot...
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Dovecot OK
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Waiting for database...
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Database OK
acme-mailcow_1       | Fri Dec 18 10:42:41 CET 2020 - Waiting for Nginx...
```

**Comment by andryyy** [comment]
You reported a bug. "It doesn't work" is not always a bug.

Reporting a bug means you post logs, explain the bug and make it - if possible - reproducible. Logs is the absolute minimum. The git diff is also useful sometimes. Just as the iptables output. Hell, at least show the running containers.

Something is wrong with your installation, I guess. "This should be fixed" doesn't help and also sounds a bit off. **What** should be fixed?

When you update mailcow and restart the stack, all kind of problems can come to light. You may have messed up the ruleset of Docker for example or created conflicting containers BEFORE you updated mailcow.

I will track this issue for information describing a bug and, IF it's a bug, I will fix it. I will not give support though, sorry. We have other channels for that. :)

**Comment by pquan** [comment]
I can confirm this is a problem with my install as well. 
It started when I updated the base ubuntu system with apt-get -y dist-upgrade
This updated docker to the following versions:
```
docker-ce 5:20.10.1~3-0~ubuntu-focal
docker-ce-cli 5:20.10.1~3-0~ubuntu-focal
docker-ce-rootless-extras 5:20.10.1~3-0~ubuntu-focal
docker-compose 1.25.0-1
```
Mailcow is presenting a wrong acme certificate as the default server cert. The wrong cert belongs to one of the "autoconfig" domains.

My Server name: mail.mydomain.com
Cert presdented: autoconfig.anotherdomain.com with anotherdomain being one of the domains hosted in mailcow.

I tried recreating the containers (docker-compose down + docker-compose up). This did not fix the issue.
Then I updated the cow using the update.sh script.
After update, the certificate presented by default changed to the correct one. The "bad one" was found as orphaned and removed.

So, maybe I was lucky and an update fixed the issue. But not sure what would happen if the cow was already updated.

The problem was caused, at the end, by a docker update. I think this should not happen.

**Comment by pquan** [comment]
> Update mailcow. That can happen when you update to Docker 20.x and forget to update mailcow.

Please elaborate on the above update of docker breaks the cow. In my understanding this should not happen. Perhaps there's an issue when the cow is restarted after some time when something happens with the certs (either expire, cannot be verified etc etc)? It seem you found a problem in the cow and fixed it and a docker-update just causes it by restarting the containers (my uneducated guess).

Please let us know what the real issue is. thank you.

**Comment by andryyy** [comment]
Docker 20.10 is fully supported after the commit that fixed the Docker 20.10 support.

I don't see any logs, no git diff, nothing. I asked for it several times. All I see are requests for support. We do support you, just not here. See the docs, thanks.
